2_filtered_results/filter_data.py

```diff
@@ -0,0 +1,395 @@
+"""Filter HuggingFace datasets by removing high force entries.
+
+This module provides functions to filter molecular datasets by removing entries
+with high RMS forces using either percentile-based or Z-score-based methods.
+The filtering helps improve dataset quality by removing potentially problematic
+high-force conformations.
+
+Command-line Arguments
+----------------------
+--data-dir : str
+    Path to the directory containing HuggingFace formatted molecular dataset.
+    This should contain dataset_info.json, state.json, and .arrow files.
+
+Examples
+--------
+Filter a dataset using Z-score method (removes entries with Z-score > 4):
+$ cd /desired/output/directory
+$ python path/to/tasks/filter_data.py --data-dir /path/to/data-raw
+
+Filter using percentile method (removes top 5% by RMS forces):
+$ cd /desired/output/directory
+$ python path/to/tasks/filter_data.py --data-dir /path/to/data-raw
+# (requires uncommenting filter_dataset_by_forces_95_percentile in main)
```

Comment:
Why not expose this as a flag to the script? Editing a script with a one-line change and then re-running the same script and command would not be my first choice